src/solver/opamCudf.ml

```diff
@@ -850,8 +838,8 @@ let extract_explanations packages cudfnv2opam reasons : explanation list =
   in
   let print_set pkgs =
     if Set.exists is_artefact pkgs then
-      if Set.exists is_opam_invariant pkgs then "(invariant)"
-      else "(request)"
+      if Set.exists is_opam_invariant pkgs then `Invariant
```

Comment:
It's worth adding a comment for the result type. It is obvious for `'Invariant` and `'Request` what they are for, but `'Formula of bool * formula` the boolean is not obvious what it is.

Reply:
fixed in 3b9d8032e